2022-07-26 23:23:15,707 ERROR: Exception on /user/b [GET] [in /home/alexey/PycharmProjects/flaskProject/venv/lib/python3.8/site-packages/flask/app.py:1455]
Traceback (most recent call last):
  File "/home/alexey/PycharmProjects/flaskProject/venv/lib/python3.8/site-packages/flask/app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/alexey/PycharmProjects/flaskProject/venv/lib/python3.8/site-packages/flask/app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/alexey/PycharmProjects/flaskProject/venv/lib/python3.8/site-packages/flask/app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/alexey/PycharmProjects/flaskProject/venv/lib/python3.8/site-packages/flask/app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/alexey/PycharmProjects/flaskProject/venv/lib/python3.8/site-packages/flask_login/utils.py", line 303, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
  File "/home/alexey/PycharmProjects/flaskProject/application/routes.py", line 70, in user
    return render_template('user.html', user=user)
  File "/home/alexey/PycharmProjects/flaskProject/venv/lib/python3.8/site-packages/flask/templating.py", line 148, in render_template
    return _render(
  File "/home/alexey/PycharmProjects/flaskProject/venv/lib/python3.8/site-packages/flask/templating.py", line 128, in _render
    rv = template.render(context)
  File "/home/alexey/PycharmProjects/flaskProject/venv/lib/python3.8/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/alexey/PycharmProjects/flaskProject/venv/lib/python3.8/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/alexey/PycharmProjects/flaskProject/application/templates/user.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/home/alexey/PycharmProjects/flaskProject/application/templates/base.html", line 31, in top-level template code
    {% block content %}{% endblock %}
  File "/home/alexey/PycharmProjects/flaskProject/application/templates/user.html", line 11, in block 'content'
    <p>{{ download_pdf.save_pdf() }}</p>
  File "/home/alexey/PycharmProjects/flaskProject/venv/lib/python3.8/site-packages/jinja2/environment.py", line 485, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'download_pdf' is undefined
2022-07-26 23:24:22,276 INFO: Microblog startup [in /home/alexey/PycharmProjects/flaskProject/application/__init__.py:32]
2022-07-26 23:24:48,943 INFO: Microblog startup [in /home/alexey/PycharmProjects/flaskProject/application/__init__.py:32]
2022-07-26 23:25:23,441 INFO: Microblog startup [in /home/alexey/PycharmProjects/flaskProject/application/__init__.py:32]
2022-07-26 23:26:30,855 INFO: Microblog startup [in /home/alexey/PycharmProjects/flaskProject/application/__init__.py:32]
